superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx

```diff
@@ -175,7 +175,13 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
       .catch(response =>
         getClientErrorObject(response).then(error => {
           addDangerToast(
-            t('ERROR: Connection failed. ') + error?.message || '',
+            error?.message
+              ? `${t('ERROR: ')}${
+                  typeof error.message === 'string'
+                    ? error.message
+                    : (error.message as Record<string, string[]>).sqlalchemy_uri
```

Comment:
The API sometimes has `message` being a dictionary of validation errors by field name, but the frontend is not able to handle that. This is just a one-off stop-gap fix which is not very scalable.

We should do a larger refactor to either explode the validation errors as multiple Error objects from the server side, handle them within `getClientErrorObject`, or more preferably, in `superset-ui/core`'s `makeApi`.

I'll add this to my todo but if anyone will be working on this area soon, feel free to take over.

cc @etr2460 since you've been working on client-side error messages.